launch/teleop.launch.py

```diff
@@ -0,0 +1,32 @@
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+import launch
+import launch_ros.actions
+
+
+def generate_launch_description():
+    joy_config = launch.substitutions.LaunchConfiguration('joy_config')
+    joy_dev = launch.substitutions.LaunchConfiguration('joy_dev')
+    config_filepath = launch.substitutions.LaunchConfiguration('config_filepath')
```

Comment:
This is my first experience with `launch.Substitutions`, so this is a question: what does this do?  Also, do we need to both declare the substitutions up here and then use `DeclareLaunchArgument` below?  It seems a bit redundant.

Reply:
This answers post by William does a pretty good job describing the difference: http://answers.ros.org/question/322874/ros2-what-is-different-between-declarelaunchargument-and-launchconfiguration/?answer=322877#post-id-322877

For better or worse, I think if `launch.Substitututions` as a mechanism for delayed expansion of a variable, and `DeclareLaunchArgument` as one possible mechanism for actually changing the value of that variable.